indico/MaKaC/common/utils.py

```diff
@@ -309,20 +309,21 @@ def getHierarchicalId(obj):
     in an hierarchical manner
     """
 
-    from MaKaC import conference
-
-    ret = obj.getId()
-    if isinstance(obj,conference.Contribution):
-        ret="%s.%s"%(obj.getConference().getId(),ret)
-    elif isinstance(obj, conference.SubContribution):
-        ret="%s.%s.%s"%(obj.getConference().getId(), obj.getContribution().getId(), ret)
-    #elif isinstance(obj, conference.DeletedObject):
-    #    ret=obj.getId().replace(':','.')
-    elif isinstance(obj, conference.Session):
-        ret="%s.s%s"%(obj.getConference().getId(), ret)
-    elif isinstance(obj, conference.SessionSlot):
-        ret="%s.s%s.%s"%(obj.getConference().getId(), obj.getSession().getId(), ret)
-    return ret
+    from indico.modules.events.contributions.models.contributions import Contribution
+    from indico.modules.events.contributions.models.subcontributions import SubContribution
+    from indico.modules.events.sessions.models.sessions import Session
+    from indico.modules.events.sessions.models.blocks import SessionBlock
```

Comment:
instead of all those imports you can simply use `db.m.MODELNAME` to access any model